src/loaders/FileLoader.js

```diff
@@ -148,6 +148,10 @@ class FileLoader extends Loader {
 
 									}
 
+								}, ( e ) => {
```

Comment:
When I think about this....Can you rewrite this by using `catch()` syntax?

Reply:
Yeah, I don't believe passing an error callback as the second argument into a "then" function is valid.

Reply:
chaining catch() will eat user-provided onprogress() exceptions too, it harms bug tracking, imagine that you get a `TypeError: failed to fetch` in onerror() while there's no network issues at all:

```js
fileloader.load(url, onload, ()=>throw 'x', console.log)
// logs `TypeError: failed to fetch`
```
so i vote no. :)

Reply:
How did you test your change? 

I would like to reproduce an error situation locally and see how the new code works.

> Yeah, I don't believe passing an error callback as the second argument into a "then" function is valid.

@ycw Do you have a reference how others have implemented this feature. TBH, I only know the `catch()` approach so far.

Reply:
tested on my own server by doing a real shutdown when streaming response body to the client, and closing stream before flushing whole asset.
  
[MDN has docs about then(f1, f2)](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/then#syntax) I also made [a quick demo](https://jsfiddle.net/y7bp96kr/) to show the critical diff between them for our case.

Reply:
I can't reproduce this with `servez`. I configured a throttling in Chrome but when I shutdown the server during a download of `FileLoader` the download is still finished to the end. 

Is there another way to force a HTTP error situation so the callback is executed?

Reply:
ok. lets craft a poor server to demo the edge case :D

<details><summary>poor_server.js</summary>

```js
// poor_server.js
require('http').createServer((req, res) => {
  if (req.url === '/a.json') {
    res.writeHead(200, { 'Content-Type': 'application/json' })
    res.write('{') // imcomplete json
    setTimeout(() => req.connection.destroy(), 2000) // disconnect for some reason
    return
  }

  res.writeHead(200, { 'Content-Type': 'text/html' })
  res.end(`
  <script type='module'>
    import * as THREE from 'https://esm.sh/three@0.163'
    const fileLoader = new THREE.FileLoader()
    fileLoader.load('a.json', ()=>{}, ()=>{}, console.log) // console won't log (<<the edge case)
  </script>`)
}).listen(9999)
``` 
</details>

1. $ node poor_server.js
2. nav to localhost:9999
3. open console
4. it should prompt `Uncaught (in promise) TypeError: network error` in ~2s